plutus-benchmark/src/Plutus/Benchmark/Knights/Sort.hs

```diff
@@ -0,0 +1,140 @@
+module Plutus.Benchmark.Knights.Sort
+        ( insertSort,
+          mergeSort,
+          quickSort
+        ) where
+
+import qualified Language.PlutusTx.Prelude as Tx
+
+{-# INLINABLE insertSort #-}
+insertSort :: (Tx.Ord a) => [a] -> [a]
+insertSort xs = foldr insertion [] xs
+
+{-# INLINABLE insertion #-}
+insertion :: (Tx.Ord a) => a -> [a] -> [a]
+insertion x [] = [x]
+insertion x (y:ys)
+        | x Tx.<= y    = x:y:ys
+        | otherwise = y:insertion x ys
+
+{-# INLINABLE mergeSort #-}
+mergeSort :: (Tx.Ord a) => [a] -> [a]
+mergeSort xs
+        = if (n <=1 ) then xs
+          else
+             (mergeList
+                ( mergeSort (take (n `div` 2) xs))
+                ( mergeSort (drop (n `div` 2) xs)))
+          where
+                n = length xs
+
+{-# INLINABLE mergeList #-}
+mergeList :: (Tx.Ord a) => [a] -> [a] -> [a]
+mergeList []   ys = ys
+mergeList xs   [] = xs
+mergeList (x:xs) (y:ys)
+        | x Tx.<= y    = x:mergeList xs (y:ys)
+        | otherwise = y:mergeList (x:xs) ys
+
+{-# INLINABLE quickSort #-}
+quickSort :: (Tx.Ord a) => [a] -> [a]
+quickSort []     = []
+quickSort (x:xs) = (quickSort [y | y<-xs, y Tx.< x]) ++ [x] ++
+                   (quickSort [y | y<-xs, y Tx.>= x])
+
+{-
```

Comment:
Ditto

Reply:
Yes, we can probably get rid of that.  I didn't quite get round to tidying everything up completely before the start of my time off, so there's quite a lot of unused stuff left over.